test/models/king_test.rb

```diff
@@ -0,0 +1,50 @@
+require 'test_helper'
+
+class KingTest < ActiveSupport::TestCase
+
+	##### TEST VALID_MOVE METHOD ######
+
+  	test "check if move is valid (invalid case, no movement)" do
```

Comment:
great work here.  The more tests around this domain logic the better.  It should really help catch logic errors as things become more complex.